components/esp_timer/src/esp_timer.c

```diff
@@ -680,6 +716,9 @@ int64_t IRAM_ATTR esp_timer_get_next_alarm_for_wake_up(void)
         timer_list_lock(dispatch_method);
         esp_timer_handle_t it = NULL;
         LIST_FOREACH(it, &s_timers[dispatch_method], list_entry) {
+            if (it == NULL) {
```

Comment:
Since LIST_FOREACH is defined as
```c++
#define	LIST_FOREACH(var, head, field)					\
	for ((var) = LIST_FIRST((head));				\
	    (var);							\
	    (var) = LIST_NEXT((var), field))
```
and `(var)` expands to `it`, looks like the `for` loop should already take care of this case?

Reply:
That's true. I guess it's just a habit to check things for NULL. I've removed the check in a2eb8a4cc5ca73fcf734fec87e5f5889a5a83546